common/src/main/java/com/wynntils/features/DialogueOptionOverrideFeature.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.google.common.collect.ImmutableList;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.features.properties.GameplayImpact;
+import com.wynntils.core.features.properties.PerformanceImpact;
+import com.wynntils.core.features.properties.Stability;
+import com.wynntils.mc.event.KeyInputEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.utils.WynnUtils;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.network.protocol.game.ServerboundSetCarriedItemPacket;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo(stability = Stability.STABLE, gameplay = GameplayImpact.SMALL, performance = PerformanceImpact.SMALL)
+public class DialogueOptionOverrideFeature extends Feature {
+    @Override
+    public MutableComponent getNameComponent() {
+        return new TranslatableComponent("feature.wynntils.dialogueOptionOverride.name");
+    }
+
+    @Override
+    protected void onInit(ImmutableList.Builder<Condition> conditions) {}
+
+    @Override
+    protected boolean onEnable() {
+        WynntilsMod.getEventBus().register(this);
+        return true;
+    }
+
+    @Override
+    protected void onDisable() {
+        WynntilsMod.getEventBus().unregister(this);
+    }
+
+    @SubscribeEvent
+    public void onDialogueKeyPress(KeyInputEvent e) {
+        if (!WynnUtils.onWorld() || e.getAction() != 1) return; // Only send packet on presses, not releases
+        if (e.getKey() - 49 == McUtils.player().getInventory().selected) { // keys 1-9 are +49 offset from hotbar
```

Comment:
does this work for keybinds?

Reply:
No it doesn't. Currently dialogue displays a [1] [2] [3] [4] set of options, which is fine for most players. Since Wynn detects these presses by the player changing the held item, do we want to remap their 1, 2, 3... buttons to send held item change packets?

If we do this, this will cause an issue (random hotbar changes) if we don't have dialogue state checks. As mentioned [here](https://github.com/Wynntils/Wynntils/pull/473), this is difficult if not impossible.

If we don't do this, users with rebound hotbars can still press their normal hotbar keys to trigger dialogue; they just won't have the override functionality.

Reply:
Yeah keybinds might be possible but would be ugly code.